pyanaconda/modules/storage/partitioning/automatic/utils.py

```diff
@@ -369,20 +369,35 @@ def schedule_volumes(storage, devices, scheme, requests, encrypted=False):
     # Convert requests into Device instances and schedule them for creation.
     #
     # Second pass, for LVs only.
+
     pool = None
+    request_thin_pool = False
+    request_thin_volume = False
+    for request in requests:
+        if request.thin_pool:
+            request_thin_pool = True
+
+        if request.thin_volume:
+            request_thin_volume = True
+
+    if request_thin_volume and not request_thin_pool:
+        # create only one default thin pool if no thin_pool is requested
+        pool = new_volume(parents=[container], thin_pool=True, grow=True)
+        storage.create_device(pool)
+
+    if pool or request_thin_pool:
+        # make sure VG reserves space for the pool to grow if needed
+        container.thpool_reserve = DEFAULT_THPOOL_RESERVE
+
+    # TODO: Ensure requests are properly ordered:
+    #  VG -> Static LV
+    #  VG -> Thin Pool - > Thin Volume
     for request in requests:
         btr = bool(scheme == AUTOPART_TYPE_BTRFS and request.btr)
         lv = bool(scheme in (AUTOPART_TYPE_LVM, AUTOPART_TYPE_LVM_THINP) and request.lv)
+        thinlp = bool(scheme == AUTOPART_TYPE_LVM_THINP and request.lv and request.thin_pool)
```

Comment:
I guess this should be just `thinp` as thin pool.

Reply:
Sure. Would you like me to fix the name right now or we wait for Jenkins feedback?